sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -51,16 +54,28 @@ abstract class LeafMathExpression(c: Double, name: String)
  * @param f The math function.
  * @param name The short name of the function
  */
-abstract class UnaryMathExpression(f: Double => Double, name: String)
+abstract class AbstractUnaryMathExpression[T, U](name: String)
```

Comment:
Hmm, because `UnaryMathExpression` has `f: Double => Double` as parameter, `BIN` can't directly extends it without modification. `BIN` has different input type (`LONG`) and output type (`STRING`) compared with other expressions (input and output: `DOUBLE`) based on `UnaryMathExpression`. For now, adding `AbstractUnaryMathExpression` maybe the simplest approach to solve this?